src/Domain.LinnApps/PurchaseOrders/PurchaseOrderService.cs

```diff
@@ -952,22 +952,22 @@ private void SetDetailFieldsForCreation(PurchaseOrderDetail detail, int newOrder
             detail.RohsCompliant = "Y";
         }
 
-        private void UpdateDeliveries(PurchaseOrderDetail purchaseOrder)
+        private void UpdateDeliveries(PurchaseOrderDetail purchaseOrder, decimal exchangeRate, int supplierId)
         {
             foreach (var delivery in purchaseOrder.PurchaseDeliveries)
             {
                 if (delivery.QuantityOutstanding > 0)
                 {
                     delivery.OrderUnitPriceCurrency = purchaseOrder.OrderUnitPriceCurrency;
                     delivery.OurUnitPriceCurrency = purchaseOrder.OurUnitPriceCurrency;
-                    delivery.NetTotalCurrency = purchaseOrder.NetTotalCurrency;
-                    delivery.VatTotalCurrency = purchaseOrder.VatTotalCurrency;
-                    delivery.DeliveryTotalCurrency = purchaseOrder.DetailTotalCurrency;
+                    delivery.NetTotalCurrency = Math.Round(delivery.OurDeliveryQty.GetValueOrDefault() * purchaseOrder.OurUnitPriceCurrency.GetValueOrDefault(), 2, MidpointRounding.AwayFromZero);
+                    delivery.VatTotalCurrency = this.purchaseOrdersPack.GetVatAmountSupplier((decimal)delivery.NetTotalCurrency, supplierId);
                     delivery.BaseOurUnitPrice = purchaseOrder.BaseOurUnitPrice;
                     delivery.BaseOrderUnitPrice = purchaseOrder.BaseOrderUnitPrice;
-                    delivery.BaseVatTotal = purchaseOrder.BaseVatTotal;
-                    delivery.BaseDeliveryTotal = Math.Round((decimal)((delivery.OurDeliveryQty * purchaseOrder.BaseOurUnitPrice.GetValueOrDefault()) + purchaseOrder.BaseVatTotal), 2);
-                    delivery.BaseNetTotal = purchaseOrder.BaseNetTotal;
+                    delivery.BaseVatTotal = Math.Round(delivery.VatTotalCurrency.GetValueOrDefault() / exchangeRate, 2, MidpointRounding.AwayFromZero);
+                    delivery.BaseNetTotal = Math.Round(delivery.OurDeliveryQty.GetValueOrDefault() * purchaseOrder.BaseOurUnitPrice.GetValueOrDefault(), 2);
```

Comment:
this (delivery.BaseNetTotal) should depend on this specific delivery price, not purchaseOrder.BaseOurUnitPrice

Reply:
wait - I'm not sure it matters since they'll both be the same value by this point anyway

Reply:
I'll update it for consistency. I did kind of wrestle with that myself and was worried about potential execution order problems but it'll probably be ok.